src/aap_eda/tasks/orchestrator.py

```diff
@@ -14,79 +14,110 @@
 
 import logging
 
+import aap_eda.tasks.activation_request_queue as requests_queue
 from aap_eda.core import models
 from aap_eda.core.enums import ActivationRequest, ActivationStatus
 from aap_eda.core.tasking import unique_enqueue
 from aap_eda.services.activation.manager import ActivationManager
-from aap_eda.tasks.activation_request_queue import (
-    list_activations,
-    peek_all,
-    pop_until,
-    push,
-)
 
-logger = logging.getLogger(__name__)
+LOGGER = logging.getLogger(__name__)
 
 
 def _manage_activation_job_id(activation_id: int) -> str:
+    """Return the unique job id for the activation manager task."""
     return f"activation-{activation_id}"
 
 
-def _manage(activation_id: int):
+def _manage(activation_id: int) -> None:
+    """Manage the activation with the given id.
+
+    It will run pending user requests or monitor the activation
+    if there are no pending requests.
+    """
     activation = models.Activation.objects.get(id=activation_id)
     has_user_requests = False
-    pending_requests = peek_all(activation_id)
+    pending_requests = requests_queue.peek_all(activation_id)
     while pending_requests:
         for request in pending_requests:
+            LOGGER.info(
+                f"Processing request {request.request} for activation "
+                f"{activation_id}",
+            )
             manager = ActivationManager(activation)
-            if request.request == ActivationRequest.START:
-                manager.start()
-            elif request.request == ActivationRequest.STOP:
-                manager.stop()
-            elif request.request == ActivationRequest.RESTART:
-                manager.restart()
-            elif request.request == ActivationRequest.DELETE:
-                manager.delete()
-            pop_until(activation_id, request.id)
+            try:
+                if request.request == ActivationRequest.START:
+                    manager.start()
+                elif request.request == ActivationRequest.STOP:
+                    manager.stop()
+                elif request.request == ActivationRequest.RESTART:
+                    manager.restart()
+                elif request.request == ActivationRequest.DELETE:
+                    manager.delete()
+            except Exception:
```

Comment:
@Alex-Izquierdo Should we make this more granular?

Reply:
@mkanoor Can you elaborate? 
As it is now and as I understand, we want to pop the request in every error. The manager leaves the activation in the right status and next cycle of the monitor will handle it if is necessary. The only missing piece is to enqueue new delayed actions accordingly to the restart policy